hooks/shapediver/useImportModelState.ts

```diff
@@ -49,38 +51,63 @@ export function useImportModelState(namespace: string) {
 			}
 
 			setIsLoading(true);
-			try {
-				const {
-					modelState: {parameters},
-				} = await sessionApi.getModelState(trimmedId);
-
-				const sessionParameters = useShapeDiverStoreParameters
-					.getState()
-					.getParameters(namespace);
-				const acceptRejectMode = Object.values(sessionParameters).some(
-					(p) => !!p?.getState().acceptRejectMode,
-				);
-
-				batchParameterValueUpdate(
-					namespace,
-					parameters,
-					!acceptRejectMode,
-				);
-
-				notifications.success({
-					message: `Model state ${trimmedId} imported successfully`,
-				});
+			const response = await exceptionWrapperAsync(
+				() => sessionApi.getModelState(trimmedId),
+				() => setIsLoading(false),
+			);
 
-				return true;
-			} catch (error) {
-				errorReporting.captureException(error);
+			if (response.error) {
+				errorReporting.captureException(response.error);
 				notifications.error({
 					message: "Failed to import model state",
 				});
 				return false;
-			} finally {
-				setIsLoading(false);
 			}
+
+			const parameters = response.data.modelState?.parameters;
+
+			if (!parameters) {
+				notifications.error({
+					message: "Model state does not contain parameter data",
+				});
+				return false;
+			}
+
+			// Get session parameters and pass to utility function
+			const sessionParameters = useShapeDiverStoreParameters
+				.getState()
+				.getParameters(namespace);
+
+			const validationResult = filterAndValidateModelStateParameters(
+				sessionParameters,
+				parameters,
+			);
+
+			if (!validationResult.hasValidParameters) {
+				const feedback = generateParameterFeedback(validationResult);
+				notifications[feedback.type]({
+					message: feedback.message,
+				});
+				return false;
+			}
+
+			await batchParameterValueUpdate(
+				namespace,
+				validationResult.validParameters,
+				!validationResult.acceptRejectMode,
```

Comment:
@LemToUp quick question: what is the reason we are skipping creating a history entry if all parameters are not using `acceptRejectMode`?

Reply:
bug found